### ISSUE TITLE
[Feature][Op] Add LogitsClip (soft capping) to GQA kernels

### ISSUE THREAD

**Issue by superAngGao** [tile-ai/TileOPs#724]
## Summary

Add logits soft capping (Gemma/Gemini style) to GQA kernels: `logits_soft_cap * tanh(score / logits_soft_cap)` applied before softmax.

Parent issue: #719
Depends on: #723 (RoPE kernels, since LogitsClip combines with RoPE)

## Op-level changes

All GQA Ops gain a `logits_soft_cap` parameter:

```python
GQAFwdOp(
    ...,
    pos_encoding_mode="ROPE_LLAMA",
    logits_soft_cap=50.0,    # None = disabled
    ...
)
```

Dispatch adds LogitsClip kernels when `logits_soft_cap is not None`:

```python
if pos_encoding_mode == "ROPE_LLAMA" and logits_soft_cap is not None:
    kernel_cls = GQARoPELogitsClipFwdKernel
elif pos_encoding_mode == "ROPE_LLAMA":
    kernel_cls = GQARoPEFwdKernel
...
```

### Op files to modify

| Op file | Change |
|---------|--------|
| `tileops/ops/gqa.py` | Add `logits_soft_cap` param and dispatch |
| `tileops/ops/gqa_decode.py` | Same |
| `tileops/ops/gqa_decode_paged.py` | Same |

### Prefill (flash_attn/)

| Kernel | File |
|--------|------|
| `GQARoPELogitsClipFwdKernel` | `rope_logits_clip_fwd.py` |
| `GQARoPELogitsClipFwdWGMMAPipelinedKernel` | same |
| `GQARoPELogitsClipFwdSplitKVKernel` | same |
| `GQARoPELogitsClipFwdSplitKVWGMMAPipelinedKernel` | same |
| `GQARoPELogitsClipBwdKernel` | `rope_logits_clip_bwd.py` |
| `GQARoPELogitsClipBwdWGMMAPipelinedKernel` | same |

### Decode (flash_decode/)

| Kernel | File |
|--------|------|
| `GQARoPELogitsClipDecodeKernel` | `gqa_rope_logits_clip_decode.py` |
| `GQARoPELogitsClipDecodeSplitKVKernel` | same |
| `GQARoPELogitsClipDecodePagedKernel` | `gqa_rope_logits_clip_decode_paged.py` |
| `GQARoPELogitsClipDecodePagedSplitKVKernel` | same |

## Models covered

Gemma 2, Gemini
